### PR TITLE
Update Privacy.js to only set cookie if preference != default

### PR DESCRIPTION
**Pull request by abhishekUpmanyu**

### CODE REVIEW

File: src/Utility/Privacy.js [hunk 1]
```diff
@@ -37,7 +37,11 @@ export default class Privacy {
     }
 
     static setAllowed(privacy_action, value) {
-        Cookie.set(this.cookieNameForAction(privacy_action), value, { expires: 365 });
+        if(value !== privacy_action.default) Cookie.set(this.cookieNameForAction(privacy_action), value, { expires: 365 });
+        else {
+            let cookie_value = Cookie.get(this.cookieNameForAction(privacy_action));
+            if(cookie_value!==undefined) Cookie.remove(this.cookieNameForAction(privacy_action));
```
Comment (baltpeter) [inline src/Utility/Privacy.js:43]: Are you sure that this is necessary? I am pretty sure that you can just delete the cookie, even if it doesn't exist.

**Review by baltpeter — CHANGES_REQUESTED** [review]
Thanks for the PR, @abhishekUpmanyu! I have left one small comment below.

Two more things:

* Please make sure that you run Prettier on the file to make sure it complies with our code style. This should happen automatically when committing if you followed the instructions in our README.
* As you can see from the [CI results](https://app.circleci.com/pipelines/github/datenanfragen/website/1168/workflows/727eeb1e-9759-4293-bf1e-c5d2d3d4b884/jobs/1919), the `privacy-controls/toggle.spec.js` test is now failing as it assumes the previous behaviour. This needs to be changed before we can merge. See [this](https://github.com/datenanfragen/website/blob/master/cypress/README.md) on how the tests work.

File: src/Utility/Privacy.js (hunk 1 above)
Comment (abhishekUpmanyu) [reply src/Utility/Privacy.js:43]: Oh, didn't think about that, I'll make the necessary change

**Comment by abhishekUpmanyu** [comment]
@baltpeter I've made the changes, and have been trying to start hugo server for running cypress tests, but I get this error while trying to run the command `hugo server`

```
Start building sites … 
ERROR 2020/10/22 14:57:24 Alias "de/e.v." contains component with a trailing space or period, problematic on Windows
Built in 47479 ms
Error: Error building site: cannot create "de/e.v.": Windows filename restriction
```

Any ideas how to get past this?

**Comment by baltpeter** [comment]
@abhishekUpmanyu Interesting. Apparently filenames may not end with a dot on Windows, surprisingly you're the first one to report that problem.

You can just delete the `"e.v."` aliases in `content/*/verein/_index.md`. That should allow the site to build.

**Comment by abhishekUpmanyu** [comment]
I've made the changes and the test that was changed now runs successfully but now I have another test failing, maybe because the cookie is not set. Please guide me a bit here

**Comment by baltpeter** [comment]
Thanks, the changes look good.

Not sure about the test failure. I was not able to reproduce that on my machine, here the test passes fine. Does the test fail when you run it locally? We are having some intermittent failures with this test (see #429) but this seems to be different. I'll try running the test a couple of times in CI to see if it is also intermittent.

**Comment by abhishekUpmanyu** [comment]
To be honest, there were issues trying to get this running on my machine, so I basically coded and pushed. I see that the tests have passed now. Will take a look at the issue you've mentioned. Cheers!

**Comment by abhishekUpmanyu** [comment]
This pull request fixes #432

**Comment by baltpeter** [comment]
Ok. Well, the problem seems to have been only temporary. The test has passed several times in a row now, so I think we can merge.

**Review by baltpeter — APPROVED** [review]
Thanks again.

**Pull request merged**